### DISCUSSION TITLE
FRAM Variant of GBA Cartridge Different CRC32

### DISCUSSION THREAD

**Discussion by WindyBison** [sanni/cartreader#520] [Need help]
Hey All,

Has anyone else encountered different crc32 values for different save variations of GBA cartridges?

For Metroid Zero Mission, I have the FRAM variant of the official cartridge and it gives a crc32 of the following:

```
OSCR HW5 V9.5

[+] Game Boy

[+] Game Boy Advance


Searching database...

Name: ZEROMISSIONE
Cart ID: BMXE
Rom Size: 16MB
Save: 256K Sram
Checksum: 1D
Version: 1.0


[+] Force Savetype


[+] 256K SRAM/FRAM


[+] Read Rom


Saving to GBA/ROM/ZEROMISSIONE/30/...
[*******************]
Internal Checksum...OK
CRC32... F7C0081E -> Not found
```

I have searched multiple GBA databases on ROM information but all of them report only the SRAM variant and a crc32 of 5C61A844 for Metroid Zero Mission.

Does anyone else know if a FRAM variant of a GBA Cart would give a different crc32?

**Comment by sanni** [comment]
Is it original or a Chinese repro? Chinese repros are often patched to save into the flashrom.

If it's original then it could be a read error. The GBA code isn't that good right now, it very susceptible to even the slightest hint of grime on the cartridge fingers. Make sure to clean the gold fingers of the cartridge with a q-tip and rubbing alcohol until the q-tip doesn't get black anymore.

Psykopat did dump the USA version with FRAM and had the standard 5c61a844 CRC32. [ See last entry here](https://datomatic.no-intro.org/index.php?page=show_record&s=23&n=1351)

**Reply by WindyBison** [reply]
Thank you for the reply!

I double cleaned all the pins and I made sure I was on version 9.4 as Psykopat. 

Here is a picture of my cart:

![Zero](https://user-images.githubusercontent.com/68243265/190687210-f6a466ba-0483-43fa-b7eb-4bc46ccee6bc.jpg)

From (little) experience I have, the cartridge to me seems legit but I might be missing something though. The Label has the stamp and the plastic seems correct. The PCB serial matches other FRAM variants of Zero mission and the chip serial seems to match BMXE. The only thing I could find is the chip seems to be of a different make and doesn't have the MX on the ROM chip from another picture I found online. Could it be this chip is different but legit? 

Picture of Other Cart from Online:
![gba_insides_003](https://user-images.githubusercontent.com/68243265/190688591-d4903094-9eaf-4d08-8921-a8e0dbcc5552.jpg)


I just borrowed my buddy's FRAM Metroid Fusion cart and it read with the correct crc32 on the reader. So, I believe the reader is good. So now I am wondering if I am just missing something with the Zero Mission Cart

**Comment by WindyBison** [comment]
One other thing that I noticed is that based on other mentioned verified Cart Dumps of Metroid Zero Mission the ROM size was 8mb, this reader made a ROM file of 16mb.

Not sure if there are different sizes of ROM that I could edit in the gba db to test out getting a good dump. 

Name: ZEROMISSIONE.gba
Size: 16777216 bytes (16 MiB)
CRC32: F7C0081E

**Reply by sanni** [reply]
Have you updated gba.txt to the latest version? https://github.com/sanni/cartreader/tree/master/sd

**Reply by WindyBison** [reply]
I was on gba.txt that came with the v9.4 portable release after I re-flashed to match Psykopat

I updated the gba.txt to latest and it looks like it read it correctly now:


```
OSCR HW5 V9.4

[+] Game Boy

[+] Game Boy Advance


Searching database...

Name: ZEROMISSIONE
Cart ID: BMXE
Rom Size: 8MB
Save: 256K Sram
Checksum: 1D
Version: 1.0


[+] Read Rom


Saving to GBA/ROM/ZEROMISSIONE/37/...
[*******************]
Internal Checksum...OK
CRC32... 5C61A844 -> Metroid - Zero Mission (USA).gba
```

I do apologize if this was a repeat discussion. I remember someone else here mentioning that writing saves with the older version of the database with Metroid Zero Mission caused some similar hiccups. I did not know it would also affect ROM reads.

**Reply by PsyK0p4T** [reply]
More details about what happened to me (and to you ^^) :

In fact I used the v9.4 files when I did that redump.
At this time, the gba.txt had a line for "Metroid Zero Mission prototype" placed before "Metroid Zero Mission USA retail". 
Both carts were using the same game serial BMXE. But rom sizes are different. 16 Mb for the proto, 8 Mb for retail cart.
So the cart reader used 16 Mb, which came first in list, dumped the cartridge, and ended with an overdump (16 mb bad dump).
The problem was reported here, github issue oppened, and sanni fixed it.
By using the current gba.txt you will not have the problem anymore ;)
